### PR TITLE
Fix odo describe for not pushed components

### LINKED ISSUE #2594: odo describe fail for not pushed components
/kind bug

## How did you run odo exactly?

`odo describe` for a not pushed component with a not pushed storage attached

## Actual behavior

```
[mrinaldas@localhost nodejs-ex]$ odo describe 
Component Name: nodejs-nodejs-ex-lhpl
Type: nodejs
Source: file://./
 ✗  deploymentconfigs.apps.openshift.io is forbidden: User "mik-dass" cannot list deploymentconfigs.apps.openshift.io in the namespace "yfvquycwjx": no RBAC policy matched
```

Add more tests related to `odo describe` but time boxing it to 1 day.

## Expected behavior
Shouldn't error out and display the info from the local config.

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind code-refactoring

**What does does this PR do / why we need it**:
Fixes odo describe for not pushed components

**Which issue(s) this PR fixes**:

Fixes  #2594

**How to test changes / Special notes to the reviewer**: `odo describe` command should not fail
```
$ odo create nodejs
$ odo storage create --size 1Gi --path  /data1
$ odo describe
```

### CODE REVIEW

File: pkg/odo/util/cmdutils.go [hunk 1]
```diff
@@ -99,17 +100,29 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 
 	}
 
+	var storages storage.StorageList
```
Comment (mik-dass) [inline pkg/odo/util/cmdutils.go]: Since we don't need these two variables outside of the below if statement, I would suggest moving them inside.

Comment (adisky) [reply pkg/odo/util/cmdutils.go]: @mik-dass Thanks for review, updated

**Comment by mohammedzee1000** [comment]
@adisky there seem to be some errors with `make validate` can you fix those?

**Comment by mohammedzee1000** [comment]
https://github.com/frapposelli/wwhrd this is the guy failing you :)

**Comment by mohammedzee1000** [comment]
/cc @kadel 
I don't see these files changed in this pr but wwhrd is complaining about it :thinking:

**Comment by adisky** [comment]
> @adisky there seem to be some errors with `make validate` can you fix those?
> 
> ```
> wwhrd check -q
> /home/travis/gopath/bin/gosec
> gosec -severity medium -confidence medium -exclude G304,G204 -quiet  ./...
> Results:
> [/home/travis/gopath/src/github.com/openshift/odo/pkg/util/config_util.go:31] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer file.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/pkg/component/component.go:1464] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer f.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/pkg/occlient/occlient.go:2869] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer f.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:90] - G306 (CWE-): Expect WriteFile permissions to be 0600 or less (Confidence: HIGH, Severity: MEDIUM)
>   > ioutil.WriteFile(filename, []byte(newContent), 0644)
> [/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:122] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer dFile.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:128] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer sFile.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:147] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer file.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/pkg/util/util.go:440] - G307 (CWE-): Deferring unsafe method "*os.File" on type "Close" (Confidence: HIGH, Severity: MEDIUM)
>   > defer file.Close()
> [/home/travis/gopath/src/github.com/openshift/odo/pkg/util/file_indexer.go:277] - G306 (CWE-): Expect WriteFile permissions to be 0600 or less (Confidence: HIGH, Severity: MEDIUM)
>   > ioutil.WriteFile(filePath, jsonData, 0666)
> Summary:
>    Files: 176
>    Lines: 27910
>    Nosec: 1
>   Issues: 9
> ```

@mohammedzee1000 I ran `make validate` in my local setup, i am not getting any error locally.

**Comment by adisky** [comment]
/hold cancel 
CI errors were fixed in #2660 and not related to this PR

**Comment by mohammedzee1000** [comment]
> /retest
> 
> Please review the [full test history](https://openshift-gce-devel.appspot.com/pr/openshift_odo/2632) for this PR and help us cut down flakes.

Thanks Rosie ;)

**Pull request merged**